src/lib/wallet/WalletConnectClient.js

```diff
@@ -0,0 +1,566 @@
+//@flow
+import { useCallback, useEffect, useState } from 'react'
+import WalletConnect from '@walletconnect/client'
+import web3Utils from 'web3-utils'
+import abiDecoder from 'abi-decoder'
+import Web3 from 'web3'
+import { first, sortBy } from 'lodash'
+import AsyncStorage from '../utils/asyncStorage'
+import { delay } from '../utils/async'
+import api from '../../lib/API/api'
+import logger from '../logger/js-logger'
+import { useSessionApproveModal } from '../../components/walletconnect/WalletConnectModals'
+import { useWallet } from './GoodWalletProvider'
+const log = logger.child({ from: 'WalletConnectClient' })
+
+//TODO:
+//7. cancel tx
+//8. edit gas
+//9. advanced edit tx values/contract call values
+//10. events
+//11. show warning if unable to decode contract call
+/**
+ * Parses the read WalletConnet URI from QR Code.
+ * If not valid, returns null.
+ * If valid, returns the WalletConnet URI.
+ * @param {string} link - receive WalletConnect URI
+ * @returns {string|null} - {link|null}
+ */
+export const readWalletConnectUri = link => {
```

Comment:
here i would like to see some class wrapping all the calls. (like good wallet class)
all the hooks should just re-call its methods during effects / in the callbacks exported and keep data in the state vars of refs.  Also hooks should be moved to './hooks'

Reply:
over kill